mesa/experimental/jupyter_viz.py

```diff
@@ -59,28 +61,106 @@ def make_model():
     def handle_change_model_params(name: str, value: any):
         set_model_parameters({**model_parameters, name: value})
 
+    @solara.component
+    def ColorCard(title, color, layout_type="Grid"):
+        with rv.Card(
+            style_=f"background-color: {color}; width: 100%; height: 100%"
+        ) as main:
+            rv.CardTitle(children=[title])
+            if layout_type == "Grid":
+                if space_drawer == "default":
+                    # draw with the default implementation
+                    make_space(model, agent_portrayal)
+                elif space_drawer:
+                    # if specified, draw agent space with an alternate renderer
+                    space_drawer(model, agent_portrayal)
+            elif layout_type == "Measure":
+                for measure in measures:
+                    if callable(measure):
+                        # Is a custom object
+                        measure(model)
+                    else:
+                        make_plot(model, measure)
+        return main
+
     # 3. Set up UI
-    solara.Markdown(name)
-    UserInputs(user_params, on_change=handle_change_model_params)
-    ModelController(model, play_interval, current_step, set_current_step, reset_counter)
-
-    with solara.GridFixed(columns=2):
-        # 4. Space
-        if space_drawer == "default":
-            # draw with the default implementation
-            make_space(model, agent_portrayal)
-        elif space_drawer:
-            # if specified, draw agent space with an alternate renderer
-            space_drawer(model, agent_portrayal)
-        # otherwise, do nothing (do not draw space)
+
+    with solara.AppBar():
+        solara.AppBarTitle(name)
+
+    grid_layout_initial = [
+        {"h": 12, "i": "0", "moved": False, "w": 5, "x": 0, "y": 0},
+        {"h": 12, "i": "1", "moved": False, "w": 5, "x": 7, "y": 0},
+    ]
+
+    # we need to store the state of the grid_layout ourselves, otherwise it will 'reset'
+    # each time we change resizable or draggable
+    grid_layout, set_grid_layout = solara.use_state(grid_layout_initial)
+
+    # render layout and plot
+
+    # jupyter
+    def render_in_jupyter():
```

Comment:
With Jupyter's limited space, could you add in fixing the columns to 2 like in line 67 in the current version. This will help ensure it renders more effectively for users if using jupyter.

Reply:
this is now added